cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/deploy/InMemoryConfigurator.java

```diff
@@ -72,11 +72,12 @@
   private final ClassLoader artifactClassLoader;
   private final String appClassName;
   private final Id.Artifact artifactId;
+  private final boolean isPreviewRun;
 
   public InMemoryConfigurator(CConfiguration cConf, Id.Namespace appNamespace, Id.Artifact artifactId,
                               String appClassName, ArtifactRepository artifactRepository,
                               ClassLoader artifactClassLoader,
-                              @Nullable String applicationName, @Nullable String configString) {
+                              @Nullable String applicationName, @Nullable String configString, boolean isPreviewRun) {
```

Comment:
This change is bad. We should introduce an overloaded constructor while keeping the old one (with preview = false). Otherwise you are necessarily changing a lot of classes.